consensus/src/sync/live/block_queue/mod.rs

```diff
@@ -304,12 +301,81 @@ impl<N: Network> BlockQueue<N> {
         );
     }
 
+    /// Attempts to insert a block into the buffer.
+    /// This may fail due to the peer who propagated the block as given in the optional `pubsub_id` parameter
+    /// already having too many blocks buffered. Blocks without a pubsub_id will always be buffered.
+    ///
+    /// At most two block per peer per block height will be buffered leaving enough room for a bock and the
+    /// skip block making it obsolete. For benign peers that is sufficient, as they will never propagate
+    /// more than that.
+    ///
+    /// ## Returns
+    /// * `true` if the block was added to the buffer
+    /// * `false` otherwise
     fn insert_block_into_buffer(&mut self, block: Block, pubsub_id: Option<N::PubsubId>) -> bool {
-        self.buffer
-            .entry(block.block_number())
-            .or_default()
-            .insert(block.hash(), (block, pubsub_id))
-            .is_some()
+        let peer_id = match pubsub_id {
+            // Blocks without a pubsub_id are requested and will always be buffered.
+            None => {
+                return self
+                    .buffer
+                    .entry(block.block_number())
+                    .or_default()
+                    .insert(block.hash(), (block, pubsub_id))
```

Comment:
```suggestion
                    .insert(block.hash(), (block, None))
```
(optional, made it clearer to me).